p2p/ipld/read.go

```diff
@@ -30,6 +31,7 @@ func RetrieveBlockData(
 ) (types.Data, error) {
 	edsWidth := len(dah.RowsRoots)
 	sc := newshareCounter(ctx, uint32(edsWidth))
+	sem := semaphore.NewWeighted(100)
```

Comment:
* Pls move the magic number to the const
* I guess the number you've chosen is simply a guess. To determine the ideal number we would need some additional analysis, but my intuition says that `100` is not enough and something like `512` would play well for us here. Althouh, I would put something like '1024' instead, but I concerned about race detector erroring with such figure, as tests may run multiple `RetreieveBlockData`. Can you play for a bit with a higher number?